apps/frontend/src/modules/quiz/slices/quiz.slice.ts

```diff
@@ -44,13 +50,29 @@ const { actions, name, reducer } = createSlice({
 		});
 		builder.addCase(getAllQuestions.fulfilled, (state, action) => {
 			state.dataStatus = DataStatus.FULFILLED;
-			state.questions = action.payload.items;
-			state.currentCategory =
-				state.questions[state.currentCategoryIndex] || null;
+			state.questionsByCategories = action.payload.items;
```

Comment:
Not sure about it, why did you change the approach?

Reply:
Those names are more descriptive. When I began developing the feature, I checked this slice to get an idea of what should be happening on the server (which entities to fetch). The name `Category` gave me an impression that a category object should be held there although in reality it was questions for the category (I do understand that there is a type annotation inside of initial state, but still). Same goas for `questions`. On itself it sounds like a single-level array of questions, while in reality this variable should hold nested arrays.